### PR TITLE
Consolidate configuration in pyproject.toml

### LINKED ISSUE #75: Merge requirements.txt with pyproject.toml
Currently the repository root has a lot of different configuration files. Consolidating them into pyproject.toml as much as possible would be nice.

For the requirements.txt case we need to:
 - change the build pipeline to actually build a python wheel file
 - install the python wheel file during the docker build step
 - migrate direct dependencies from requirements file to pyproject.toml
 - define entypoint in pyproject.toml and use that entrypoint in docker container (necessary because we don't simply copy all soures any more)

### PR DESCRIPTION
**Pull request by MRuecklCC**
Resolves #75.

### CODE REVIEW

**Comment by MRuecklCC** [comment]
All the docker build steps still fail as there is no more requirements.txt file. Also the pipeline now needs poetry to build the wheel file. This functionality still needs to be taken from the meta-qs repository.

**Comment by RMeissnerCC** [comment]
> 

So I will not merge this, but wait till you fix the pipeline :)

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,7 @@
+; Unfortunately flake8 doesn't support having configuration in pyproject.toml:
+; https://github.com/PyCQA/flake8/issues/234
+[flake8]
+ignore = E203, E266, E501, W503, F403, F401
```
Comment (RMeissnerCC) [inline .flake8:4]: You could check out in MetaQS, how I configured flake8 there directly in the pre-commit file. This way, we have one file less :)

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -1,9 +1,11 @@
 repos:
-  - repo: https://github.com/ambv/black
```
Comment (RMeissnerCC) [inline .pre-commit-config.yaml:2]: What is the reason behind dropping this?

File: tests/e2e/e2e_test.py [hunk 3]
```diff
@@ -1,25 +1,15 @@
...
-import os
-import time
 from json import JSONDecodeError
 
 import pytest
 import requests
 
 from lib.settings import SKIP_E2E_TESTS
-
-if "PRE_COMMIT" in os.environ:
```
Comment (RMeissnerCC) [inline tests/e2e/e2e_test.py:11]: When running the tests from pre commit hook, I needed those lines. Are they now not needed anymore? Due to the __init__.py?

File: tests/integration/features_integration_test.py [hunk 4]
```diff
@@ -675,7 +673,7 @@ def test_javascript():
 
     html = {
         "html": "<script src='/xlayer/layer.php?uid='></script>"
-                "<script href='some_test_javascript.js'></script>",
+        "<script href='some_test_javascript.js'></script>",
```
Comment (RMeissnerCC) [inline tests/integration/features_integration_test.py:676]: Is black activated in your pre-commit hooks? This looks - different.

File: tests/e2e/e2e_test.py (hunk 3 above)
Comment (MRuecklCC) [reply tests/e2e/e2e_test.py:11]: yes, tests should now behave identical during pre_commit and normal execution!

File: .pre-commit-config.yaml (hunk 2 above)
Comment (MRuecklCC) [reply .pre-commit-config.yaml:2]: Uhm, for some reason it didn't work for me (there was a dependency clash with one of blacks dependencies.) And as black was listed as part of the poetry development dependencies we can use that black instead!

File: tests/integration/features_integration_test.py (hunk 4 above)
Comment (MRuecklCC) [reply tests/integration/features_integration_test.py:676]: It is now, but might not have been when i pushed this :/

File: .pre-commit-config.yaml (hunk 2 above)
Comment (RMeissnerCC) [reply .pre-commit-config.yaml:2]: Personally, I would prefer to drop black out of the poetry dev dependencies and just have it in the pre-commit hook, where it is called.

File: .flake8 (hunk 1 above)
Comment (MRuecklCC) [reply .flake8]: done. now configured via command line args in pre-commit setup.

**Pull request merged**